public/static/app/common/ApiUtil.js

```diff
@@ -3,10 +3,15 @@ mciModule.factory('ApiUtil', function($http) {
   // TODO Use $httpProvider (tech-debt ticket required)
   return {
     httpGetter: function(base) {
-      return function(endpointTpl, endpointTplParams, httpParams) {
+      var realBase = (
```

Comment:
Having a nested ternary like this is pretty hard to read. Could you use a switch case instead?

Reply:
Why do you think so? It's basically
```javascript
cond ? then :
cond ? then :
else
```
Moreover, `switch` doesn't return a value and requires `break`s.